### ISSUE TITLE
Getting Internal Server Error with latest of boto

### ISSUE THREAD

**Issue by heldersepu** [getmoto/moto#8247]
## how to reproduce the issue

```
import boto3
from moto import mock_aws


class Test_CUSTOM_ENABLE_S3_POLICY:
    @mock_aws
    def test_success2(self):
        client = boto3.client("s3", region_name="us-east-1")
        client.create_bucket(Bucket="test")
        client.put_bucket_policy(Bucket="test", Policy='{}')
        assert 1 == 1
```

happens with latest, if I hard code to older it works:
```
pytest
boto3 == 1.35.37
botocore == 1.35.37
moto >= 5.0.16.dev5
```

## what actually happens
the test takes a lot longer as it is reaching to AWS
> E           botocore.exceptions.ClientError: An error occurred (500) when calling the PutBucketPolicy operation (reached max retries: 4): Internal Server Error

## version of Moto
```
Name: moto
Version: 5.0.18
```
and
`Successfully installed boto3-1.35.45 botocore-1.35.45`

**Comment by bpandola** [comment]
I think this may be an issue with the latest version of `botocore` (see https://github.com/boto/botocore/issues/3284).

**Comment by zsaltys** [comment]
+1 seeing the same issue, breaking a lot of stuff

**Comment by bpandola** [comment]
This should be fixed by updating to `botocore>=1.35.47`.

**Comment by heldersepu** [comment]
confirmed, it works with `botocore>=1.35.47`
